File: wyca_test/scripts/listener.py
#!/usr/bin/env python
import roslib
import rospy
from std_msgs.msg import String
from geometry_msgs.msg import Twist
from sound_play.libsoundplay import SoundClient
import sys


def callback(cmd_RC):
    #rospy.loginfo("Received a /cmd_vel message!")

    linearv=cmd_RC.linear.x
    angularv=cmd_RC.angular.z
    rospy.loginfo("Linear velocity x: [%f]"%(linearv))
    rospy.loginfo("Angular velocity x: [%f]"%(angularv))

    if linearv == 0 and angularv == 0:
	soundhandle.say("keylo. Stop.", voice)
	rospy.sleep(6)
    elif linearv > 0 and angularv == 0:
	soundhandle.say("KEYLO. Moving. Forward.", voice)
	rospy.sleep(6)
    elif linearv < 0 and angularv == 0:
	soundhandle.say("KEYLO. Moving. Backward.", voice)
	rospy.sleep(1)
    elif linearv == 0 and angularv > 0:
	soundhandle.say("KEYLO. Turning. Right.", voice)
	rospy.sleep(6)
    elif linearv == 0 and angularv < 0:
	soundhandle.say("KEYLO. Turning. Left.", voice)
	rospy.sleep(6)
'''
    while True:
        angularv1=angularv
        linearv1=linearv
        linearv=cmd_RC.linear.x
        angularv=cmd_RC.angular.z
        rospy.loginfo("Linear velocity x: [%f]"%(linearv))
        rospy.loginfo("Angular velocity x: [%f]"%(angularv))	   
        if linearv1 <> linearv or angularv1 <> angularv:
	   break
'''

def listener():
    #Initialize ROS
    rospy.init_node('listener', anonymous=True)
    # The start message
    soundhandle.say("Ready. For. Start.", voice)
    rospy.sleep(5)
    #Subscribe to the /cmd_vel topic
    rospy.Subscriber("/cmd_vel", Twist, callback)

    # spin() simply keeps python from exiting until this node is stopped
    rospy.spin()


if __name__ == '__main__':
    # Set the default TTS voice to use
    voice = rospy.get_param("~voice", "voice_don_diphone")
    #Create the sound client object
    soundhandle = SoundClient()
    listener()

File: wyca_test/src/listener.cpp
#include "ros/ros.h"
#include "std_msgs/String.h"
#include "geometry_msgs/Twist.h"
#include <sound_play/sound_play.h>
#include <unistd.h>


void cmdRCCallback(const geometry_msgs::Twist::ConstPtr& cmd_RC)
{
	float linearv=cmd_RC->linear.x;
	float angularv=cmd_RC->angular.z;
	//std::cout << "Linear velocity x: "<< cmd_RC->linear.x << " \t Angular velocity z: "<< cmd_RC->angular.z << std::endl;
	ROS_INFO("Linear velocity x: [%f]",linearv);
        ROS_INFO("Angular velocity z: [%f]",angularv);
	sound_play::SoundClient sc;

	if (linearv == 0 and angularv == 0)
	{		
		sc.say("keylo. Stop..","voice_don_diphone");
		sleep(6);
	}
 	else if (linearv > 0 and angularv == 0)
	{
				
		sc.say("KEYLO. Moving. Forward.","voice_don_diphone");
		sleep(6);
	}
  	else if (linearv < 0 and angularv == 0)
	{		
		sc.say("KEYLO. Moving. Backward.","voice_don_diphone");
		sleep(6);
	}
        else if (linearv == 0 and angularv > 0)
	{			
		sc.say("KEYLO. Turning. Right.","voice_don_diphone");
		sleep(6);
	}
        else if (linearv == 0 and angularv < 0)
	{
					
		sc.say("KEYLO. Turning. Left.","voice_don_diphone");
		sleep(6);
	}

	
}
void sleepok(int t, ros::NodeHandle &nh)
{
   if (nh.ok())
   sleep(t);
}

int main(int argc, char **argv)
{
	//Initialize ROS
	ros::init(argc, argv, "listener");
        //NodeHandle is the main access point to communications with the ROS system.
	ros::NodeHandle nh;
	//Create the sound client object
        sound_play::SoundClient sc;
        sleepok(1, nh);
	// Set the default TTS voice to use
	sc.say("Ready. For. Start.","voice_don_diphone");
	//Subscribe to the /cmd_vel topic 
	ros::Subscriber sub = nh.subscribe("cmd_vel", 100, cmdRCCallback);
	
    while (ros::ok())
    {
      ros::spin();
    }

    return 0;
}







